### ISSUE TITLE
Incorrect results returned when using JOIN statement

### ISSUE THREAD

**Issue by chenglinLiang** [duckdb/duckdb#12634]
### What happens?

When I use the JOIN statement to combine two tables, DuckDB returns incorrect results. Additionally, DuckDB produces different results depending on the inserted data.

### To Reproduce

``` sql
CREATE TABLE t1(c0 VARCHAR UNIQUE DEFAULT(-1891621750));
CREATE TABLE t2(c0 VARCHAR, PRIMARY KEY(c0));
```
**Case one**
```sql
INSERT INTO t1(c0) VALUES (DEFAULT);
INSERT INTO t2(c0) VALUES (1708578242), ('|㵝'), (-1.89162175E9);
SELECT t1.c0,t1.rowid,t2.c0 FROM t1 LEFT JOIN t2 ON t1.c0 >= t2.c0;
```
```
┌─────────────┬───────┬────┐
│     c0      │ rowid │ c0 │
├─────────────┼───────┼────┤
│ -1891621750 │ 0     │    │
└─────────────┴───────┴────┘
```
```sql
SELECT t1.c0,t1.rowid,t2.c0 FROM t1 RIGHT JOIN t2 ON t1.c0 >= t2.c0;
```
```
┌────┬───────┬───────────────┐
│ c0 │ rowid │      c0       │
├────┼───────┼───────────────┤
│    │       │ -1891621750.0 │
│    │       │ 1708578242    │
│    │       │ |㵝           │
└────┴───────┴───────────────┘
```
```sql
SELECT t1.c0,t1.rowid,t2.c0 FROM t1 FULL JOIN t2 ON t1.c0 >= t2.c0;
```
```
┌─────────────┬───────┬───────────────┐
│     c0      │ rowid │      c0       │
├─────────────┼───────┼───────────────┤
│ -1891621750 │ 0     │ -1891621750.0 │
│             │       │ |㵝           │
│             │       │ 1708578242    │
└─────────────┴───────┴───────────────┘
```

**Case two**
```sql
INSERT INTO t1(c0) VALUES (DEFAULT);
INSERT INTO t2(c0) VALUES (-1.89162175E9);
SELECT t1.c0,t1.rowid,t2.c0 FROM t1 LEFT JOIN t2 ON t1.c0 >= t2.c0;
```
```
┌─────────────┬───────┬───────────────┐
│     c0      │ rowid │      c0       │
├─────────────┼───────┼───────────────┤
│ -1891621750 │ 0     │ -1891621750.0 │
└─────────────┴───────┴───────────────┘
```
```sql
SELECT t1.c0,t1.rowid,t2.c0 FROM t1 RIGHT JOIN t2 ON t1.c0 >= t2.c0;
```
```sql
┌────┬───────┬───────────────┐
│ c0 │ rowid │      c0       │
├────┼───────┼───────────────┤
│    │       │ -1891621750.0 │
└────┴───────┴───────────────┘
```
```sql
SELECT t1.c0,t1.rowid,t2.c0 FROM t1 FULL JOIN t2 ON t1.c0 >= t2.c0;
```
```
┌─────────────┬───────┬───────────────┐
│     c0      │ rowid │      c0       │
├─────────────┼───────┼───────────────┤
│ -1891621750 │ 0     │ -1891621750.0 │
└─────────────┴───────┴───────────────┘
```

### OS:

Ubuntu 20.04.6 x64

### DuckDB Version:

v0.5.1 (7c11132)

### DuckDB Client:

Command line

### Full Name:

Personal

### Affiliation:

Personal

### What is the latest build you tested with? If possible, we recommend testing with the latest nightly build.

I have not tested with any build

### Did you include all relevant data sets for reproducing the issue?

Yes

### Did you include all code required to reproduce the issue?

- [X] Yes, I have

### Did you include all relevant configuration (e.g., CPU architecture, Python version, Linux distribution) to reproduce the issue?

- [X] Yes, I have

**Comment by szarnyasg** [comment]
Hi @chenglinLiang, thanks for opening this issue. Unfortunately, we do not support version 0.5.1 anymore. Can you please try to upgrade to the latest version, v1.0.0?

I cannot reproduce the results with the latest version:

Case 1:

```sql
D CREATE TABLE t1(c0 VARCHAR UNIQUE DEFAULT(-1891621750));
Run Time (s): real 0.001 user 0.000632 sys 0.000156
D CREATE TABLE t2(c0 VARCHAR, PRIMARY KEY(c0));
Run Time (s): real 0.001 user 0.000907 sys 0.000090
D INSERT INTO t1(c0) VALUES (DEFAULT);
Run Time (s): real 0.001 user 0.001130 sys 0.000353
D INSERT INTO t2(c0) VALUES (1708578242), ('|㵝'), (-1.89162175E9);
Run Time (s): real 0.002 user 0.000706 sys 0.000219
D SELECT t1.c0,t1.rowid,t2.c0 FROM t1 LEFT JOIN t2 ON t1.c0 >= t2.c0;
┌─────────────┬───────┬─────────┐
│     c0      │ rowid │   c0    │
│   varchar   │ int64 │ varchar │
├─────────────┼───────┼─────────┤
│ -1891621750 │     0 │ NULL    │
└─────────────┴───────┴─────────┘
Run Time (s): real 0.001 user 0.001608 sys 0.000456
D SELECT t1.c0,t1.rowid,t2.c0 FROM t1 RIGHT JOIN t2 ON t1.c0 >= t2.c0;
┌─────────┬───────┬───────────────┐
│   c0    │ rowid │      c0       │
│ varchar │ int64 │    varchar    │
├─────────┼───────┼───────────────┤
│ NULL    │  NULL │ 1708578242    │
│ NULL    │  NULL │ |㵝           │
│ NULL    │  NULL │ -1891621750.0 │
└─────────┴───────┴───────────────┘
Run Time (s): real 0.001 user 0.001206 sys 0.000402
D SELECT t1.c0,t1.rowid,t2.c0 FROM t1 FULL JOIN t2 ON t1.c0 >= t2.c0;
┌─────────────┬───────┬───────────────┐
│     c0      │ rowid │      c0       │
│   varchar   │ int64 │    varchar    │
├─────────────┼───────┼───────────────┤
│ NULL        │  NULL │ 1708578242    │
│ NULL        │  NULL │ |㵝           │
│ NULL        │  NULL │ -1891621750.0 │
│ -1891621750 │     0 │ NULL          │
└─────────────┴───────┴───────────────┘
Run Time (s): real 0.002 user 0.001683 sys 0.000314
```

Case two:

```sql
D CREATE TABLE t1(c0 VARCHAR UNIQUE DEFAULT(-1891621750));
Run Time (s): real 0.001 user 0.000478 sys 0.000154
D CREATE TABLE t2(c0 VARCHAR, PRIMARY KEY(c0));
Run Time (s): real 0.000 user 0.000783 sys 0.000085
D INSERT INTO t1(c0) VALUES (DEFAULT);
Run Time (s): real 0.002 user 0.001238 sys 0.000491
D INSERT INTO t2(c0) VALUES (-1.89162175E9);
Run Time (s): real 0.000 user 0.000642 sys 0.000210
D SELECT t1.c0,t1.rowid,t2.c0 FROM t1 LEFT JOIN t2 ON t1.c0 >= t2.c0;
┌─────────────┬───────┬─────────┐
│     c0      │ rowid │   c0    │
│   varchar   │ int64 │ varchar │
├─────────────┼───────┼─────────┤
│ -1891621750 │     0 │ NULL    │
└─────────────┴───────┴─────────┘
Run Time (s): real 0.001 user 0.001508 sys 0.000426
D SELECT t1.c0,t1.rowid,t2.c0 FROM t1 RIGHT JOIN t2 ON t1.c0 >= t2.c0;
┌─────────┬───────┬───────────────┐
│   c0    │ rowid │      c0       │
│ varchar │ int64 │    varchar    │
├─────────┼───────┼───────────────┤
│ NULL    │  NULL │ -1891621750.0 │
└─────────┴───────┴───────────────┘
Run Time (s): real 0.001 user 0.001649 sys 0.000290
D SELECT t1.c0,t1.rowid,t2.c0 FROM t1 FULL JOIN t2 ON t1.c0 >= t2.c0;
┌─────────────┬───────┬───────────────┐
│     c0      │ rowid │      c0       │
│   varchar   │ int64 │    varchar    │
├─────────────┼───────┼───────────────┤
│ NULL        │  NULL │ -1891621750.0 │
│ -1891621750 │     0 │ NULL          │
└─────────────┴───────┴───────────────┘
Run Time (s): real 0.001 user 0.001786 sys 0.000316
```

**Comment by chenglinLiang** [comment]
Thanks for your prompt response! I have switched to the latest version and obtained the same results as you, which is very encouraging to see that this issue might have been fixed. In v0.5.1, this error caused by JOIN will cause data loss. Could you please confirm if there are any mentions of fixing similar issues in the subsequent release notes?

**Comment by szarnyasg** [comment]
I tried with old versions of DuckDB, and this issue has been fixed since 0.6.0. I cannot find a specific entry on with the terms `FULL`/`OUTER`/`FULL OUTER` join in the [release notes](https://github.com/duckdb/duckdb/releases/tag/v0.6.0) but I'm quite confident the issues has been resolved.
